### PR TITLE
Fix directional light camera calculation

### PR DESCRIPTION
**Pull request by YuqiaoZhang**
Fix directional light camera calculation

### CODE REVIEW

**Comment by hyv1001** [comment]
Please give a clear and concise description of what the problem is.

**Comment by YuqiaoZhang** [comment]
It may be incorrect to calcuate the Intersection of the "frustum_bounding_box" and the "scene_bounding_box". 
Some objects, which are outside the "frustum_bounding_box" but closer to the light source, may cast shadow as well.

File: engine/source/runtime/function/render/source/vulkan_manager/misc/misc.cpp [hunk 1]
```diff
@@ -207,20 +207,26 @@ namespace Pilot
...
         {
-            glm::vec3 corner        = extents * g_BoxOffset[i] + center;
-            glm::vec4 corner_with_w = m * glm::vec4(corner.x, corner.y, corner.z, 1.0);
-            corner                  = glm::vec3(corner_with_w.x / corner_with_w.w,
-                               corner_with_w.y / corner_with_w.w,
-                               corner_with_w.z / corner_with_w.w);
-
-            min = glm::min(min, corner);
-            max = glm::max(max, corner);
+            glm::vec3 corner_before = extents * g_BoxOffset[i] + center;
```
Comment (hyv1001) [inline engine/source/runtime/function/render/source/vulkan_manager/misc/misc.cpp:216]: please format code with clang-format

Comment (hyv1001) [inline engine/source/runtime/function/render/source/vulkan_manager/misc/misc.cpp:216]: please format code with clang-format

**Pull request merged**